docs/pages/database-access/guides/mongodb-self-hosted.mdx

```diff
@@ -0,0 +1,348 @@
+  hostname, run:
+
+  ```bash
+  $ tctl auth sign --format=mongodb --host=mongo.example.com --out=mongo --ttl=2190h
+  ```
+
+  (!docs/pages/includes/database-access/ttl-note.mdx!)
+
+  The command will create 2 files: `mongo.cas` with Teleport's certificate
+  authority and `mongo.crt` with generated certificate and key pair. You will
+  need these files to enable mutual TLS on your MongoDB server.
+  </TabItem>
+  <TabItem label="Replica set">
+  When connecting to a MongoDB replica set, sign certificates for each member
+  using the hostnames they're accessible at.
+
+  For example, if the first member is accessible at `mongo1.example.com` and
+  the second at `mongo2.example.com`, run:
+
+  ```bash
+  $ tctl auth sign --format=mongodb --host=mongo1.example.com --out=mongo1 --ttl=2190h
+  $ tctl auth sign --format=mongodb --host=mongo2.example.com --out=mongo2 --ttl=2190h
+  ```
+
+  (!docs/pages/includes/database-access/ttl-note.mdx!)
+
+  Each command will create 2 files: `mongo1.cas`/`mongo2.cas` with Teleport's
+  certificate authority and `mongo1.crt`/`mongo2.crt` with generated certificate
+  and key pair. You will need these files to enable mutual TLS on your MongoDB
+  servers.
+  </TabItem>
+</Tabs>
+
+## Configure MongoDB users
+
+Users authenticating with client certificates must be created in the `$external`
+MongoDB authentication database.
+
+MongoDB treats the entire `Subject` line of the client certificate as a username.
+When connecting to a MongoDB server, say as a user `alice`, Teleport will sign
+an ephemeral certificate with `CN=alice` subject.
+
+To create this user in the database you can run the following `mongo` shell
+command:
+
+```js
+db.getSiblingDB("$external").runCommand(
+  {
+    createUser: "CN=alice",
+    roles: [
+      { role: "readWriteAnyDatabase", db: "admin" }
+    ]
+  }
+)
+```
+
+Update the [roles](https://docs.mongodb.com/manual/tutorial/manage-users-and-roles/)
+accordingly to grant the user appropriate database permissions.
+
+See [Use X.509 Certificates to Authenticate Clients](https://docs.mongodb.com/manual/tutorial/configure-x509-client-authentication/)
+in MongoDB documentation for more details.
+
+## Configure MongoDB TLS
+
+Add the following section to your `mongod.conf` configuration file to enable
+mutual TLS. Use the files generated above by the `tctl auth sign` command:
+
+<Tabs>
+  <TabItem label="MongoDB 3.6 - 4.2">
+  ```yaml
+  net:
+    ssl:
+      mode: requireSSL
+      PEMKeyFile: /etc/certs/mongo.pem
+      CAFile: /etc/certs/mongo.cas
+  ```
+  </TabItem>
+  <TabItem label="MongoDB 4.2+">
+  ```yaml
+  net:
+    tls:
+      mode: requireTLS
+      certificateKeyFile: /etc/certs/mongo.pem
+      CAFile: /etc/certs/mongo.cas
+  ```
+  </TabItem>
+</Tabs>
+
+When configuring a replica set, make sure to do it for each member and use
+secrets generated for the particular server.
+
+<Admonition type="note" title="">
```

Comment:
Stylistic comment: I think `title` is optional, you can either leave it out or add whatever you want. We often will just add the `type`. 

Thanks for using these!